lib/node/failure_detector.ml

```diff
@@ -79,26 +79,27 @@ let update_peer_status node peer status =
       (Printf.sprintf "Failed to find peer with address %s:%d in node peer list"
          peer.address.address peer.address.port)
 
-let create_message node message (recipient : Peer.t) =
+let create_message node message (recipients : Peer.t list) =
   Message.
     {
       category = Failure_detection;
       id = -1;
+      timestamp = Unix.gettimeofday ();
       sender = Client.address_of !node;
-      recipient = recipient.address;
+      recipients = List.map (fun p -> p.Peer.address) recipients;
       payload = Encoding.pack bin_writer_message message;
     }
 
-let send_message message node (recipient : Peer.t) =
-  let message = create_message node message recipient in
-  Client.send_to node message
+let send_message message node (recipients : Peer.t list) =
+  let message = create_message node message recipients in
+  Networking.send_to node message
 
-let send_ping_to node peer = send_message Ping node peer
+let send_ping_to node peer = send_message Ping node [peer]
```

Comment:
Hum, I guess there is still some work to do within the `handle_message` function, where we do:
```ocaml
let _ = List.map (send_ping_request_to node) pingers in
```